frontends/web/src/routes/device/bitbox02/sdcardcheck.tsx

```diff
@@ -14,80 +14,68 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
+import { FunctionComponent, useEffect, useState } from 'react';
 import { translate, TranslateProps } from '../../../decorators/translate';
 import { Dialog, DialogButtons } from '../../../components/dialog/dialog';
 import { apiGet } from '../../../utils/request';
 import { Button, ButtonLink } from '../../../components/forms';
 
-interface SDCardCheckProps {
+type SDCardCheckProps = {
     deviceID: string;
 }
 
-interface State {
-    sdCardInserted?: boolean;
-}
-
-
 type Props = SDCardCheckProps & TranslateProps;
 
-class SDCardCheck extends Component<Props, State> {
-  state = {} as State;
+const SDCardCheck: FunctionComponent<Props> = ({ t, deviceID, children }) => {
 
-  public componentDidMount() {
-    this.check();
-  }
+  const [sdCardInserted, setSdCardInserted] = useState<boolean | undefined>(undefined);
+
+  useEffect(() => {
+    check();
+    // no dependencies passed - only gets executed once on mount
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  private check = () => {
-    apiGet('devices/bitbox02/' + this.props.deviceID + '/check-sdcard')
-      .then(inserted => this.setState({ sdCardInserted: inserted }));
+  const check = () => {
+    apiGet('devices/bitbox02/' + deviceID + '/check-sdcard')
+      .then(inserted => setSdCardInserted(inserted));
   };
```

Comment:
Could you try with our custom hook `useLoad` instead?

Reply:
See comment above